src/contexts/authContext.js

```diff
@@ -0,0 +1,65 @@
+import React, { createContext, useEffect, useReducer } from 'react';
+
+const AuthContext = createContext();
+
+const initialState = {
+  isAuthenticated: false,
+  user: null,
+  token: null,
+};
+
+const reducer = (state, action) => {
```

Comment:
how about move the reducer out of this file? it would be better if we have something like:
src
|__contexts
    |__auth
        |__reducer.js
         __index.js

Reply:
Moved!